frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx

```diff
@@ -0,0 +1,212 @@
+/* eslint jest/expect-expect: ["error", { "assertFunctionNames": ["expect", "assertBanner"] }] */
+import { Route } from "react-router";
+
+import { renderWithProviders, screen } from "__support__/ui";
+import type { State } from "metabase-types/store";
+
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import {
+  createMockTokenFeatures,
+  createMockUser,
+} from "metabase-types/api/mocks";
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+
+import { AppBanner } from "./AppBanner";
+
+interface SetupOpts {
+  shouldRenderPastDueBanner: boolean;
+  shouldRenderUnpaidBanner: boolean;
+  shouldRenderDatabasePromptBanner: boolean;
+}
+
+function setup({
+  shouldRenderPastDueBanner,
+  shouldRenderUnpaidBanner,
+  shouldRenderDatabasePromptBanner,
+}: SetupOpts) {
+  renderWithProviders(<Route path="*" component={AppBanner} />, {
+    initialRoute: "/",
+    storeInitialState: createStateForConfig({
+      shouldRenderPastDueBanner,
+      shouldRenderUnpaidBanner,
+      shouldRenderDatabasePromptBanner,
+    }),
+    withRouter: true,
+  });
+}
+
+function createStateForConfig({
+  shouldRenderPastDueBanner,
+  shouldRenderUnpaidBanner,
+  shouldRenderDatabasePromptBanner,
+}: SetupOpts): State {
+  let isAdmin = false;
+  const tokenStatus = {
+    status: "Token is valid.",
+  };
+  const features = {
+    sso: false,
+  };
+
+  if (shouldRenderPastDueBanner) {
+    tokenStatus.status = "past-due";
+    isAdmin = true;
+  }
+
+  if (shouldRenderUnpaidBanner) {
+    tokenStatus.status = "unpaid";
+    isAdmin = true;
+  }
+
+  if (shouldRenderDatabasePromptBanner) {
+    isAdmin = true;
+    features.sso = true;
+    setupDatabasesEndpoints([createSampleDatabase()]);
+  }
+
+  const state = createMockState({
+    currentUser: createMockUser({ is_superuser: isAdmin }),
+    settings: createMockSettingsState({
+      "token-features": createMockTokenFeatures(features),
+      "token-status": tokenStatus,
+    }),
+  });
+
+  return state;
+}
+
+describe("AppBanner", () => {
+  it.each([
+    {
+      shouldRenderPastDueBanner: false,
```

Comment:
As before, I think there should be no setup like this, but feel free to fix this in another PR

Reply:
This test should never be here in the first place. This PR was in the middle of waiting for the previous 2 to be merged that's why I didn't rebase this one off the integration branch right away to avoid wasting CI cycles.

The new tests will be added in a followed-up PR as discussed.